### PR TITLE
Enable ergonomic brand checks (`#priv in`) by default

### PR DESCRIPTION
**Pull request by nicolo-ribaudo**
| Q                        | A 
| ------------------------ | ---
| Patch: Bug Fix?          |
| Major: Breaking Change?  |
| Minor: New Feature?      | Yes
| Tests Added + Pass?      | Yes
| Documentation PR Link    | https://github.com/babel/website/pull/2556
| Any Dependency Changes?  |
| License                  | MIT

@ljharb is going to ask for Stage 4 at this week's meeting ([agenda](https://github.com/tc39/agendas/blob/master/2021/07.md)).

We were already planning 7.15.0, but it would be very nice to get this PR in (if the feature gets Stage 4).

**EDIT:** Stage 4 confirmed :tada:

<a href="https://gitpod.io/#https://github.com/babel/babel/pull/13554"></a>

### CODE REVIEW

File: packages/babel-parser/src/parser/expression.js [hunk 1]
```diff
@@ -1153,16 +1153,12 @@ export default class ExpressionParser extends LValParser {
...
         const value = this.state.value;
         node = this.parsePrivateName();
         if (this.match(tt._in)) {
-          this.expectPlugin("privateIn");
           this.classScope.usePrivateName(value, node.start);
-        } else if (this.hasPlugin("privateIn")) {
-          this.raise(this.state.start, Errors.PrivateInExpectedIn, value);
         } else {
-          throw this.unexpected(start);
+          this.raise(this.state.start, Errors.PrivateInExpectedIn, value);
```
Comment (JLHwung) [inline packages/babel-parser/src/parser/expression.js:1161]: Here we no longer throw at the private identifier, is it intentional?

Comment (nicolo-ribaudo) [reply packages/babel-parser/src/parser/expression.js:1161]: It's GitHub's diff not rendering it correctly; this was the deleted else-if branch.

**Review by hzoo — APPROVED** [review]

File: packages/babel-parser/src/parser/expression.js (hunk 1 above)
Comment (JLHwung) [reply packages/babel-parser/src/parser/expression.js:1161]: I don't think so. Before this PR `this.state.start` is accessed _before_ `parsePrivateName`. In this PR it is _after_. So the error positions are changed from

```
"throws": "Unexpected token (4:11)"
```

to

```
"SyntaxError: Private names are only allowed in property accesses (`obj.#x`) or in `in` expressions (`#x in obj`). (4:13)"
```

Comment (nicolo-ribaudo) [reply packages/babel-parser/src/parser/expression.js:1161]: The old code was
https://github.com/babel/babel/blob/bfcf783834cbe293ca540e84c1ad285ed2e728e2/packages/babel-parser/src/parser/expression.js#L1163
I didn't write this line, I just deleted the `hasPlugin` check and removed the now unreachable `else` clause.

But yeah, it's true that the old position without the plugin was better.

**Review by JLHwung — APPROVED** [review]

**Comment by ljharb** [comment]
This proposal is now stage 4

 - https://github.com/tc39/proposal-private-fields-in-in/issues/7
 - https://github.com/tc39/proposals/commit/0e01f7e01ecb7af256a574ea73c1f1794da1b4f2

**Review by existentialism — APPROVED** [review]

**Comment by ljharb** [comment]
yay!

**Comment by nicolo-ribaudo** [comment]
We'll release 7.15 this week, hopefully tomorrow!

**Pull request merged**